api/src/main/java/org/apache/iceberg/types/Type.java

```diff
@@ -92,6 +93,10 @@ default boolean isListType() {
     return false;
   }
 
+  default boolean isVariantType() {
```

Comment:
This is only used in a deprecated method, do we have any other reason to add this? I think it probably doesn't need to be apart of the type interface. We could always just check if the type is VARIANT

Reply:
Yeah. This is more a helper function instead of checking if the type is VARIANT.  Actually it will be used in some other places later.

Reply:
I removed this for now. We can add it back when it's getting used later.